### PR TITLE
Php router

### PR DESCRIPTION
**Pull request by jstirling91**
This allows the integration tests to use the pretty urls

### CODE REVIEW

File: htdocs/router.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<?php
+error_log("URL:  " . $_SERVER['REQUEST_URI']);
```
Comment (driusan) [inline htdocs/router.php]: This seems to be for debugging.

File: htdocs/router.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+if (preg_match('#^/bootstrap/(.*)#', $_SERVER["REQUEST_URI"])){
+	// Bootstrap is not a Loris module, don't rewrite it.
+	// RewriteRule ^bootstrap/(.*) bootstrap/$1 [L]
+	return false;
```
Comment (driusan) [inline htdocs/router.php]: What does return false do?

File: htdocs/router.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+} else if (preg_match('#^/([0-9]{6,6})/$#', $_SERVER["REQUEST_URI"])) {
+	// Redirect /CandID/ to visit label list
+ 	// RewriteRule ^([0-9]{6,6})/$ /main.php?test_name=timepoint_list&candID=$1
+ 	$getParams = explode("/", $_SERVER["REQUEST_URI"]);
+	$_REQUEST["test_name"] = "timepoint_list";
+	$_REQUEST['candID'] = $getParams[1];
```
Comment (driusan) [inline htdocs/router.php]: I think that, in general, you need to set both $_REQUEST and $_GET ($_REQUEST is the union of $_GET and $_POST in PHP).. you should probably make a helper function to do that and just call that.

**Comment by driusan** [comment]
This is failing because of PHPCS (mostly it looks like errors that can be fixed by PHPCBF).

But regardless, how do you use/test this?

File: htdocs/router.php (hunk 2 above)
Comment (jstirling91) [inline htdocs/router.php]: Return false makes it so that no action is taken on the request, and executes it as direct path request

**Comment by jstirling91** [comment]
To run it using php built-in web server run the command 
```php -S localhost:8000 -t htdocs/ htdocs/router.php```

**Comment by driusan** [comment]
I'm trying to use this and it seems to be more sensitive to the number of slashes than the mod_rewrite rules.. ie if a module loads "//datadict/" instead of "/datadict/", it tries to load /datadict/main.php instead of the datadict module.

**Pull request merged**